### PR TITLE
Simplify Star.lhs

### PR DESCRIPTION
**Pull request by sid-kap**
Attempts to make the code easier to understand

(Disclaimer: I haven't yet verified that this compiles/produces the correct output.)

### CODE REVIEW

File: web/gallery/Star.lhs [hunk 1]
```diff
@@ -18,22 +18,27 @@ height: 400
...
 >   where
->     xCoord1 = round . fst . unp2 . fst
->     ps = zip xs (reverse ys)
->     (xs, ys) = pts n
+>     getColor :: Int -> Colour
+>     getColor i = colors !! (p1 `mod` 2)
+>
+>     makeLine :: Colour -> P2 Double -> P2 Double -> Diagram B
+>     makeLine color p q =
+>       p \~\~ q
```
Comment (byorgey) [inline web/gallery/Star.lhs]: Nothing should need to be escaped, this should be `p ~~ q` rather than `p \~\~ q`.

File: web/gallery/Star.lhs [hunk 2]
```diff
@@ -18,22 +18,27 @@ height: 400
...
->         & shaftStyle %~ lw thin . lc (colors !! ((xCoord1 p) `mod` 2)))
->   (fst p) (snd p) | p <- ps]
+> quarter :: Int -> Diagram B
+> quarter n = mconcat [ makeLine (getColor i) (p2 (i, 0)) (p2 (0, n - i)) | i <- [0..n] ]
 >   where
->     xCoord1 = round . fst . unp2 . fst
->     ps = zip xs (reverse ys)
->     (xs, ys) = pts n
+>     getColor :: Int -> Colour
+>     getColor i = colors !! (p1 `mod` 2)
```
Comment (byorgey) [inline web/gallery/Star.lhs]: Where is `p1` defined?  Should this be `i` instead?

**Comment by byorgey** [comment]
Looks good to me in general.  I left a few specific comments inline.

**Comment by byorgey** [comment]
Cool, this looks nice.  Interesting to look at some diagrams code I wrote a *long* time ago and see how much my style has changed.

There seems to still be a type error, it can't decide whether `i` should be `Int` or `Double`.

**Comment by sid-kap** [comment]
Hmm, looks like Travis is OOMing again :(

Also, on my local machine, I'm getting lots of errors that look like: 
```
Error while interpreting
> example = square 1 # fc red ||| square 1 # fc blue
<interactive>:2:1:
    Couldn't match expected type ‘V2’
                with actual type ‘V (IO (QDiagram Rasterific V2 Double Any))’
    In the expression: example
    In an equation for ‘e_1’: e_1 = example
    In the expression:
        (let e_1 = example in e_1) ::
          IO (QDiagram Rasterific V2 Double Any)


error: diagram could not be rendered
```
when it's trying to build the rst files. Has this been happening to you as well?

**Comment by byorgey** [comment]
No, that's strange.  I vaguely seem to remember that happening to someone, somewhere, at some point in the past, but I don't remember what the problem was.

**Pull request closed without merging**